src/main/java/com/pingcap/tikv/types/DurationType.java

```diff
@@ -0,0 +1,120 @@
+/*
+ *
+ * Copyright 2017 PingCAP, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.pingcap.tikv.types;
+
+import com.pingcap.tikv.codec.CodecDataInput;
+import com.pingcap.tikv.codec.CodecDataOutput;
+import com.pingcap.tikv.codec.InvalidCodecFormatException;
+import com.pingcap.tikv.meta.TiColumnInfo;
+import com.pingcap.tikv.row.Row;
+
+import java.time.LocalDateTime;
+
+public class DurationType extends IntegerType {
+  static DurationType of(int tp) {
+    return new DurationType(tp);
+  }
+
+  private DurationType(int tp) {
+    super(tp);
+  }
+
+  DurationType(TiColumnInfo.InternalTypeHolder holder) {
+    super(holder);
+  }
+
+  @Override
+  public Object decodeNotNull(int flag, CodecDataInput cdi) {
+    if (flag == VARINT_FLAG) {
+      long nanoSec = IntegerType.readVarLong(cdi);
+      return nanoSec / 1000000;
+    } else if (flag == INT_FLAG) {
+      long nanoSec = IntegerType.readLong(cdi);
+      return nanoSec / 1000000;
+    } else {
+      throw new InvalidCodecFormatException("Invalid Flag type for Time Type: " + flag);
+    }
+  }
+
+  /**
+   * decode a value from cdi to row per tp.
+   *
+   * @param cdi source of data.
+   * @param row destination of data
+   * @param pos position of row.
+   */
+  public void decode(CodecDataInput cdi, Row row, int pos) {
```

Comment:
The difference btw decode and decodeNotNull is, decode suppose to be left in parent class which deal with null and then it will call decodeNotNull. DecodeNotNull is to be implemented in subclass to deal with detailed decoding. 
1. why needs to be implemented here
2. if you do need it, please make sure null is handled.